pkgs/applications/editors/moe/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchurl
+, lzip, ncurses
+}:
+
+with stdenv.lib;
+stdenv.mkDerivation rec{
+
+  name = "moe-${version}";
+  version = "1.9";
+
+  src = fetchurl{
+    url = "http://ftp.gnu.org/moe/${name}.tar.lz";
+    sha256 = "1wsfzy0iia0c89wnx1ilzw54wqcmlp2nz8mkpvc393z0zagrx48q";
+  };
+
+  buildInputs = [ lzip ncurses ];
+
+  meta = {
+    description = "A small, 8-bit clean editor";
+    longDescription = ''
+      GNU moe is a powerful, 8-bit clean, console text editor for ISO-8859 and
+      ASCII character encodings. It has a modeless, user-friendly interface,
+      online help, multiple windows, unlimited undo/redo capability, unlimited
+      line length, unlimited buffers, global search/replace (on all buffers at
+      once), block operations, automatic indentation, word wrapping, file name
+      completion, directory browser, duplicate removal from prompt histories,
+      delimiter matching, text conversion from/to UTF-8, romanization, etc.
+    '';
+    homepage = http://www.gnu.org/software/moe/;
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ AndersonTorres ];
+    platforms = platforms.unix;
```

Comment:
On macOS (aka darwin), build fails with:

```
c++  -O2 -c -o window_vector.o window_vector.cc
window_vector.cc:1033:21: error: call to member function 'insert' is ambiguous
      { year_string.insert( 0U, 1, '0' + ( year % 10 ) ); year /= 10; }
        ~~~~~~~~~~~~^~~~~~
```

A possible work-around is to declare a less general platform.

Reply:
Is it triggered only in clang?

Reply:
Either that or this code is not compiled on Linux.